rust-src/concordium_base/src/transactions.rs

```diff
@@ -2389,6 +2428,45 @@ pub mod construct {
         }
     }
 
+    /// A transaction that is prepared to be signed.
+    /// The serde instance serializes the structured payload and skips
+    /// serializing the encoded one.
+    #[derive(Debug, Clone, SerdeSerialize)]
+    #[serde(rename_all = "camelCase")]
+    pub struct PreAccountTransactionV1 {
+        pub header: TransactionHeaderV1,
+        /// The payload.
+        pub payload: Payload,
+        /// The encoded payload. This is already serialized payload that is
+        /// constructed during construction of the prepared transaction
+        /// since we need it to compute the cost.
+        #[serde(skip_serializing)]
+        pub encoded: EncodedPayload,
+        /// Hash of the transaction to sign.
+        pub hash_to_sign: hashes::TransactionSignHash,
+    }
```

Comment:
Would put sender and sponsor signatures in here to support adding sponsor signatures prior to adding sender signatures.
```suggestion
    pub struct PreAccountTransactionV1 {
        pub header: TransactionHeaderV1,
        /// The payload.
        pub payload: Payload,
        /// The encoded payload. This is already serialized payload that is
        /// constructed during construction of the prepared transaction
        /// since we need it to compute the cost.
        #[serde(skip_serializing)]
        pub encoded: EncodedPayload,
        /// Hash of the transaction to sign.
        pub hash_to_sign: hashes::TransactionSignHash,
        pub sender_signature: Option<AccountTransactionSignature>,
        pub sponsor_signature: Option<AccountTransactionSignature>,
    }
```